external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -844,12 +866,16 @@ impl PassResult {
     }
 }
 
+/// Run the compiler until a desired pass is reached. If `no_fail` argument is true, it is
+/// guaranteed that this function will return `PassResult` (and not `Diagnostics`) as part of it
+/// result.
 fn run(
     compilation_env: &mut CompilationEnv,
     pre_compiled_lib: Option<&FullyCompiledProgram>,
     cur: PassResult,
     until: Pass,
     mut result_check: impl FnMut(&PassResult, &CompilationEnv),
+    no_fail: bool,
```

Comment:
Instead of adding it as a flag, could we add it to the compilation env? Then `check_diags_at_or_above_severity` could consider the flag there.